src/networkProviders/resources.ts

```diff
@@ -0,0 +1,206 @@
+     */
+    public timestamp: number = 0;
+
+    /**
+     * The block timestamp.
+     */
+    public round: number = 0;
+
+    /**
+     * The block timestamp.
+     */
+    public epoch: number = 0;
+
+    /**
+     * Constructs a configuration object from a HTTP response (as returned by the provider).
+     */
+    static fromHttpResponse(payload: any): BlockOnNetwork {
+        let blockOnNetwork = new BlockOnNetwork();
+
+        blockOnNetwork.raw = payload;
+        blockOnNetwork.shard = Number(payload["shard"]) ?? 0;
+        blockOnNetwork.nonce = BigInt(payload["nonce"] ?? 0);
+        blockOnNetwork.hash = payload["hash"] ?? "";
+        blockOnNetwork.previousHash = payload["prevBlockHash"] ?? payload["prevHash"] ?? "";
+        blockOnNetwork.timestamp = Number(payload["timestamp"] ?? 0);
+        blockOnNetwork.round = Number(payload["round"] ?? 0);
+        blockOnNetwork.epoch = Number(payload["epoch"] ?? 0);
+
+        return blockOnNetwork;
+    }
+}
+
+export class BlockCoordinates {
+    nonce: bigint = 0n;
+    hash: string = "";
+    rootHash?: string;
+    constructor(init?: Partial<BlockCoordinates>) {
+        Object.assign(this, init);
+    }
+    static fromHttpResponse(payload: any): BlockCoordinates {
+        const result = new BlockCoordinates();
+        const value = payload["blockInfo"] || {};
+
+        result.nonce = value["nonce"] || 0n;
+        result.hash = value["hash"] || "";
+        result.rootHash = value["rootHash"] || "";
+
+        return result;
+    }
+}
+
+export class AccountStorageEntry {
+    raw: Record<string, any> = {};
+    address: Address = Address.empty();
+    key: string = "";
+    value: string = "";
+    constructor(init?: Partial<AccountStorageEntry>) {
+        Object.assign(this, init);
+    }
+
+    static fromHttpResponse(payload: any, key: string): AccountStorageEntry {
+        const result = new AccountStorageEntry();
+        const value = payload["value"] || "";
+
+        result.raw = payload;
+        result.key = key;
+        result.value = BytesValue.fromHex(value).toString();
+
+        return result;
+    }
+}
+
+export class AwaitingOptions {
+    pollingIntervalInMilliseconds: number = DEFAULT_ACCOUNT_AWAITING_POLLING_TIMEOUT_IN_MILLISECONDS;
+    timeoutInMilliseconds: number = DEFAULT_ACCOUNT_AWAITING_TIMEOUT_IN_MILLISECONDS;
+    patienceInMilliseconds: number = DEFAULT_ACCOUNT_AWAITING_PATIENCE_IN_MILLISECONDS;
+}
+
+export class TransactionCostEstimationResponse {
+    raw: Record<string, any> = {};
+    gasLimit: number = 0;
+    status: TransactionStatus = TransactionStatus.createUnknown();
+
+    static fromHttpResponse(payload: any): TransactionCostEstimationResponse {
+        const result = new TransactionCostEstimationResponse();
+        result.raw = payload;
+        result.gasLimit = payload["txGasUnits"] ?? 0;
+        result.status = new TransactionStatus("");
+
+        return result;
+    }
+}
+
+export class GetBlockArguments {
```

Comment:
I think this is removed from sdk-py. Does it make sense to drop it here, as well, and also in specs?